crates/engine/tree/src/tree/mod.rs

```diff
@@ -2262,8 +2274,43 @@ where
         // prewarming
         let caches = ProviderCacheBuilder::default().build_caches();
         let cache_metrics = CachedStateMetrics::zeroed();
-        let state_provider =
-            CachedStateProvider::new_with_caches(state_provider, caches, cache_metrics);
+        let state_provider = CachedStateProvider::new_with_caches(
+            state_provider,
+            caches.clone(),
+            cache_metrics.clone(),
+        );
+
+        // TODO: move this logic somewhere better?
+        for tx in block.body().transactions() {
```

Comment:
you could wrap the entire thing in `pool.scope(|s| )` to avoid cloning but then it would block at scope end

Reply:
scoping could be useful because we _will_ need to make sure all tasks are complete when we are doing cross-block caching + prewarming (basically using scope as a waitgroup). But I think it would make this fn body a bit of a mess, and we could accomplish a similar thing by using a `RwLock`